### PR TITLE
EVG-2495 Add setup/teardown for task group groups to agent

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: agent/agent.go [hunk 1]
```diff
@@ -303,7 +308,23 @@ func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) {
...
+}
+
+func (a *Agent) runPostGroupCommands(ctx context.Context, tc *taskContext, taskGroup *model.TaskGroup) {
+	grip.Info("Running post-group commands")
+	a.killProcs(tc)
+	defer a.killProcs(tc)
+	var cancel context.CancelFunc
+	ctx, cancel = a.withCallbackTimeout(ctx, tc)
+	defer cancel()
+	var err error
```
Comment (tychoish) [inline agent/agent.go]: why not just use := below?

File: agent/agent.go [hunk 2]
```diff
@@ -303,7 +308,23 @@ func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) {
...
+	err = a.runCommands(ctx, tc, taskGroup.TeardownGroup, false)
+	if err != nil {
+		grip.Errorf("Error running post-task command: %v", err)
+	} else {
+		grip.Info("Finished running post-group commands")
```
Comment (tychoish) [inline agent/agent.go]: I'm inclined to just bake this into conditional logging/errorswrap statements

File: agent/agent.go [hunk 3]
```diff
@@ -303,7 +308,23 @@ func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) {
...
 		}
-		a.killProcs(tc)
+		grip.Info("Finished running post-task commands")
+	}
+}
+
+func (a *Agent) runPostGroupCommands(ctx context.Context, tc *taskContext, taskGroup *model.TaskGroup) {
+	grip.Info("Running post-group commands")
+	a.killProcs(tc)
+	defer a.killProcs(tc)
```
Comment (tychoish) [inline agent/agent.go]: I think we need to kill procs inbetween tasks even after task groups run. 

we can make this configurable now or later.

Comment (bsamek) [reply agent/agent.go]: Sharing procs is EVG-2671. If you're saying that we should kill procs between tasks even if postGroupTaskCommands doesn't run, this is handled already by `runTask`.

File: agent/agent.go (hunk 1 above)
Comment (bsamek) [reply agent/agent.go]: fixed

File: agent/agent.go (hunk 2 above)
Comment (bsamek) [reply agent/agent.go]: fixed

File: agent/task.go [hunk 4]
```diff
@@ -107,14 +107,16 @@ func (a *Agent) startTask(ctx context.Context, tc *taskContext, complete chan<-
...
-		tc.logger.Execution().Errorf("error creating task directory: %s", err)
-		complete <- evergreen.TaskFailed
-		return
+	if tc.runGroupSetup {
+		tc.taskDirectory, err = a.createTaskDirectory(tc)
+		if err != nil {
+			tc.logger.Execution().Errorf("error creating task directory: %s", err)
+			complete <- evergreen.TaskFailed
+			return
+		}
```
Comment (tychoish) [inline agent/task.go]: we talked in person to see if we could merge this, somehow, with the "taskPreCommand" but if that isn't obvious, it's not *that* weird, I think

**Pull request merged**